bofire/surrogates/surrogate.py

```diff
@@ -44,27 +45,75 @@ def predict(self, X: pd.DataFrame) -> pd.DataFrame:
             Xt[c] = pd.to_numeric(Xt[c], errors="raise")
         # predict
         preds, stds = self._predict(Xt)
+        # set up column names
+        pred_cols = []
+        sd_cols = []
+        for featkey in self.outputs.get_keys(CategoricalOutput):
+            pred_cols = pred_cols + [
+                f"{featkey}_{cat}_prob"
+                for cat in self.outputs.get_by_key(featkey).categories  # type: ignore
+            ]
+            sd_cols = sd_cols + [
+                f"{featkey}_{cat}_sd"
+                for cat in self.outputs.get_by_key(featkey).categories  # type: ignore
+            ]
+        for featkey in self.outputs.get_keys(ContinuousOutput):
+            pred_cols = pred_cols + [f"{featkey}_pred"]
+            sd_cols = sd_cols + [f"{featkey}_sd"]
         # postprocess
         predictions = pd.DataFrame(
             data=np.hstack((preds, stds)),
-            columns=["%s_pred" % featkey for featkey in self.outputs.get_keys()]
-            + ["%s_sd" % featkey for featkey in self.outputs.get_keys()],
+            columns=pred_cols + sd_cols,
         )
+        # append predictions for categorical cases
+        for feat in self.outputs.get(CategoricalOutput):
+            predictions.insert(
+                loc=0,
+                column=f"{feat.key}_pred",
+                value=predictions.filter(regex=f"{feat.key}(.*)_prob")
+                .idxmax(1)
+                .str.replace(f"{feat.key}_", "")
+                .str.replace("_prob", "")
+                .values,
+            )
+            predictions.insert(
+                loc=1,
+                column=f"{feat.key}_sd",
+                value=0.0,
+            )
         # validate
         self.validate_predictions(predictions=predictions)
         # return
         return predictions
 
     def validate_predictions(self, predictions: pd.DataFrame) -> pd.DataFrame:
-        expected_cols = [
-            f"{key}_{t}" for key in self.outputs.get_keys() for t in ["pred", "sd"]
-        ]
+        expected_cols = []
+        check_columns = []
```

Comment:
We are composing this kind of lists add a lof of different places, a few examples are:
- `data_models/domain/features.py`
- `data_models/strategies/predictives/predictive.py`

Can we create a utility method to create the correct col names etc. and move it into utils and just call it from different places. This will also making testing much easier.

Reply:
Let us put it into a new file under utils, as you want it ...